server/storage/endpoint/key_path.go

```diff
@@ -103,3 +105,33 @@ func gcSafePointServicePath(serviceID string) string {
 func MinResolvedTSPath() string {
 	return path.Join(clusterPath, minResolvedTS)
 }
+
+// KeySpaceServiceSafePointPrefix returns the prefix of given service's service safe point.
+// Prefix: /key_space/gc_safepoint/{space_id}/service/
+func KeySpaceServiceSafePointPrefix(spaceID string) string {
+	return path.Join(keySpaceSafePointPrefix, spaceID, "service") + "/"
+}
+
+// KeySpaceGCSafePointPath returns the gc safe point's path of the given key-space.
+// Path: /key_space/gc_safepoint/{space_id}/gc
+func KeySpaceGCSafePointPath(spaceID string) string {
+	return path.Join(keySpaceSafePointPrefix, spaceID, keySpaceGCSafePointSuffix)
+}
+
+// KeySpaceServiceSafePointPath returns the path of given service's service safe point.
+// Path: /key_space/gc_safepoint/{space_id}/service/{service_id}
```

Comment:
Can we know the endpoint types from `service_id`? such as it's from `tidb`, `raw_kv client`, `cdc` or others?

Reply:
I don't think we can tell by `service_id` alone, 
for example, both `TiDB` and `raw_kv's gc_worker` may use `gc_worker` as their `service_id`

Reply:
So... Is it different in space_id? like `cdc_xxx`, `tidb_xxx`, `client_xxxx` . I would like to know who use the path.

Reply:
currently, we use `spaceID` to differentiate key spaces, things like
`default_rawkv`, `default_txnkv`, `default_tidb`
`serviceID` is used to identify the specific service under that `KeySpace`, like
`cdc`, `br`, etc.

`tidb cdc` and `rawkv cdc` may have the same `serviceID` but different `spaceID`s
`rawkv cdc` and `rawkv br` have the same `spaceID` but different `serviceID`s